spark-cassandra-connector/src/main/scala/org/apache/spark/sql/cassandra/PushDown.scala

```diff
@@ -0,0 +1,204 @@
+package org.apache.spark.sql.cassandra
+
+import com.datastax.spark.connector.cql.TableDef
+import org.apache.spark.sql.catalyst.expressions._
+import org.apache.spark.sql.sources
+import org.apache.spark.sql.sources.Filter
+
+/**
+ * Calculate pushdown predicates for a given table and Catalyst [[Expression]]s
+ */
+class PredicatePushDown (predicates: Seq[Expression], tableDef: TableDef)
+  extends PushDown(predicates: Seq[Expression], tableDef: TableDef) {
+
+  def isEqualTo(predicate: Expression): Boolean = predicate.isInstanceOf[EqualTo]
+
+  def isIn(predicate: Expression) : Boolean = predicate.isInstanceOf[In] || predicate.isInstanceOf[InSet]
+
+  def isRangeComparison(predicate: Expression) : Boolean = predicate match {
+    case _: LessThan           => true
+    case _: LessThanOrEqual    => true
+    case _: GreaterThan        => true
+    case _: GreaterThanOrEqual => true
+    case _                     => false
+  }
+
+  override def isSingleColumn(predicate: Expression) : Boolean  =
+    predicate.references.size == 1 && super.isSingleColumn(predicate)
+
+  /** Returns the only column name referenced in the predicate */
+  def predicateColumnName(predicate: Expression) : String = {
+    require(predicate.references.size == 1, "Given predicate is not a single column predicate: " + predicate)
```

Comment:
A better format would be this so the user knows what field is of issue

s"'predicate' $predicate must be a single column"